### PR TITLE
Ref #4596: Create jdbc tests for generated keys and other headers

### LINKED ISSUE #4596: Expand JDBC test coverage
The following scenarios from the [documentation](https://camel.apache.org/components/3.20.x/jdbc-component.html) are not covered in Camel Quarkus:

1. [Generated keys](https://camel.apache.org/components/3.20.x/jdbc-component.html#_generated_keys)
2. [Using named parameters](https://camel.apache.org/components/3.20.x/jdbc-component.html#_using_named_parameters)
3. Polling databes every time interval - part of the [samples](https://camel.apache.org/components/3.20.x/jdbc-component.html#_sample_polling_the_database_every_minute).
4. Move data between datasources - part of the [samples](https://camel.apache.org/components/3.20.x/jdbc-component.html#_sample_move_data_between_data_sources).
 
Because JDBC extension leverages quarkus extension (see the [documentaion](https://quarkus.io/guides/datasource)) it would be nice to cover major features - it is optional.

1.  Cover different db types for jdbc (based on the [doc](https://quarkus.io/guides/datasource#jdbc-datasource-2)) - DB2, Derby, H2, MariaDB, Microsoft SQL Server, MySQL, Oracle and PostgreSQL.

### PR DESCRIPTION
**Pull request by ldrozdo**
Partially fixes #4596:
1. Generated keys
2. Message headers ( except for CamelJdbcParameters header, which will be done in the next PR)

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java [hunk 1]
```diff
@@ -27,12 +28,7 @@
 import jakarta.annotation.PostConstruct;
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
-import jakarta.ws.rs.Consumes;
-import jakarta.ws.rs.GET;
-import jakarta.ws.rs.POST;
-import jakarta.ws.rs.Path;
-import jakarta.ws.rs.PathParam;
-import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.*;
```
Comment (zbendhiba) [inline integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java:31]: I guess this one is related to your IDE configuration. 
Can you please revert importing all the package?

**Review by zbendhiba — CHANGES_REQUESTED** [review]

File: integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java [hunk 2]
```diff
@@ -109,4 +107,29 @@ public String getSpeciesByIdWithDefinedType(@PathParam("id") String id) throws E
     public String executeStatement(String statement) throws Exception {
         return template.requestBody("jdbc:camel-ds", statement, String.class);
     }
+
+    @Path("/generated-keys/rows")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List generatedKeysRows() throws Exception {
+        return template.requestBodyAndHeader("direct://get-generated-keys",
+                "insert into camelsGenerated (species) values ('Camelus testus'), ('Camelus redhatus')",
```
Comment (aldettinger) [inline integration-tests/jdbc/src/main/java/org/apache/camel/quarkus/component/jdbc/CamelResource.java:116]: Shouldn't we avoid to refer to company names ?

File: integration-tests/jdbc/src/test/java/org/apache/camel/quarkus/component/jdbc/CamelJdbcTest.java [hunk 3]
```diff
@@ -16,14 +16,17 @@
...
+
 import io.quarkus.test.common.QuarkusTestResource;
 import io.quarkus.test.h2.H2DatabaseTestResource;
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.*;
```
Comment (zbendhiba) [inline integration-tests/jdbc/src/test/java/org/apache/camel/quarkus/component/jdbc/CamelJdbcTest.java:28]: Same comment here.

**Review by aldettinger — APPROVED** [review]
I filed a suggestion. Beyond, looks good :+1:

**Review by zbendhiba — APPROVED** [review]

**Review by zhfeng — APPROVED** [review]

**Pull request merged**